com.unity.render-pipelines.core/Tests/Editor/RemoveComponentTests.cs

```diff
@@ -0,0 +1,152 @@
+                .SetName("RemovesMultipleAdditionalDatas"),
+            new TestCaseData(typeof(Apple), new Type[] {typeof(Banana), typeof(AdditionalBanana), typeof(Apple), typeof(AdditionalApple)}, new Type[] {typeof(Banana), typeof(AdditionalBanana)})
+                .SetName("OnlyRemoveItsAdditionalData")
+        };
+
+        [Test, TestCaseSource(nameof(s_RemoveComponentTestCaseDatas))]
+        public void RemoveComponent([DisallowNull] Type componentToRemove, [DisallowNull] Type[] componentsToAdd, [DisallowNull] Type[] expectedComponents)
+        {
+            Assert.True(Property.RemoveComponent(m_GameObject, componentToRemove, componentsToAdd, expectedComponents));
+        }
+
+        static TestCaseData[] s_RemoveAdditionalDataComponentTestCaseDatas =
+        {
+            new TestCaseData(typeof(AdditionalBanana), new Type[] {typeof(Banana), typeof(AdditionalBanana) }, Array.Empty<Type>())
+                .SetName("RemoveSingleAdditionalData"),
+            new TestCaseData(typeof(AdditionalBananaColor), new Type[] {typeof(Banana), typeof(AdditionalBanana), typeof(AdditionalBananaColor)}, Array.Empty<Type>())
+                .SetName("RemovesMultipleAdditionalDatas"),
+           new TestCaseData(typeof(AdditionalBananaColor), new Type[] {typeof(Banana), typeof(Banana), typeof(AdditionalBanana), typeof(AdditionalBananaColor)}, Array.Empty<Type>())
+                .SetName("RemovesMultipleComponent"),
+            new TestCaseData(typeof(AdditionalApple), new Type[] {typeof(Banana), typeof(AdditionalBanana), typeof(Apple), typeof(AdditionalApple)}, new Type[] {typeof(Banana), typeof(AdditionalBanana)})
+                .SetName("OnlyRemoveItsAdditionalData")
+        };
+
+        [Test, TestCaseSource(nameof(s_RemoveAdditionalDataComponentTestCaseDatas))]
+        public void RemoveAdditionalDataComponent([DisallowNull] Type componentToRemove, [DisallowNull] Type[] componentsToAdd, [DisallowNull] Type[] expectedComponents)
+        {
+            Assert.True(Property.RemoveAdditionalComponent(m_GameObject, componentToRemove, componentsToAdd, expectedComponents));
+        }
+
+        static TestCaseData[] s_TryGetComponentsToRemoveTestCaseDatas =
+         {
+            new TestCaseData(typeof(FruitBasket))
+                .Returns(new string[] { "Apple", "Banana" })
+                .SetName("Multiple"),
+            new TestCaseData(typeof(AdditionalBanana))
+                .Returns(new string[] {"Banana"})
+                .SetName("Single"),
+            new TestCaseData(typeof(WaterMelon))
+                .Returns(Array.Empty<string>())
+                .SetName("Empty"),
+        };
+
+        private GameObject m_GameObject;
+        [SetUp]
+        public void SetUp()
+        {
+            m_GameObject = new("RemoveComponentTestsGO");
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            GameObject.DestroyImmediate(m_GameObject);
+        }
```

Comment:
for readability, I would perhaps move the SetUp/TearDown at bottom or top of this class, not in between tests